File: python/word-count/word_count.py
from collections import Counter
import re

SPLIT_PATTERN = re.compile(r'[\W_]+', re.UNICODE)

def words(sentence):
  words = SPLIT_PATTERN.split(unicode(sentence, 'utf-8').lower())
  return [word for word in words if word != '']

def word_count(sentence):
  return dict(Counter(words(sentence)))

File: elixir/word-count/word_count.exs
defmodule Words do
  defp string_to_word_list(str) do
    String.downcase(str) |> String.split(~r/(_|[^\w\p{Pd}])+/u, trim: true)
  end

  defp update_map(word, map) do
    Map.update(map, word, 1, &(&1 + 1))
  end

  def count(sentence) do
    string_to_word_list(sentence) |> Enum.reduce(%{}, &update_map/2)
  end
end